### ISSUE TITLE
Do not include log4j in compile dependencies

### ISSUE THREAD

**Issue by oniseijin** [symphonyoss/symphony-java-client#81]
Currently, the dependency tree is as follows: 

+--- org.slf4j:slf4j-api:1.7.21 -> 1.7.25
|    +--- org.slf4j:slf4j-log4j12:1.7.21
|    |    +--- org.slf4j:slf4j-api:1.7.21 -> 1.7.25
|    |    \--- log4j:log4j:1.2.17

Please don't include slf4j-log4j as a compile time dependency. Place it as a "runtime" dependency. Scoping details at https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html 

Basically add <scope>runtime</scope> to you pom for org.slf4j:slf4j-log4j12. This is messing with the ability to use log4j 2.x in a project. API is correct to leave as compile scope.

**Comment by ftbb** [comment]
commit 8efb28e10dbd1c56e1bbef0376083b8d43ba26f1

Note:  I will be switching to logback over the next few commits.
